common.js

```diff
@@ -103,7 +103,11 @@ function setopts (self, pattern, options) {
   if (process.platform === "win32")
     self.root = self.root.replace(/\\/g, "/")
 
-  self.cwdAbs = makeAbs(self, self.cwd)
+  // TODO: is an absolute `cwd` supposed to be resolved against `root`?
+  // e.g. { cwd: '/test', root: __dirname } === path.join(__dirname, '/test')
+  self.cwdAbs = isAbsolute(self.cwd) ? self.cwd : makeAbs(self, self.cwd)
```

Comment:
This seems like a reasonable fix.  We do need `makeAbs` to mount onto the root, if one is specified, so that `/a/*` maps onto `{root}/a/b`.  But using cwd and root together is just... kinda weird.